src/foam/nanos/fs/File.js

```diff
@@ -267,7 +283,15 @@ foam.CLASS({
       name: 'authorizeOnDelete',
       javaCode: `
         AuthService auth = (AuthService) x.get("auth");
-        if ( ! auth.check(x, "file.remove." + getId()) ) {
+        Boolean hasPermission = auth.check(x, "file.remove." + getId());
+
+        // find a file owner
+        // creating a file uses businessID for the file owner
+        DAO businessDAO = (DAO) x.get("businessDAO");
+        Business fileOwner = (Business) businessDAO
+          .find(EQ(Business.ID, getOwner()));
```

Comment:
should check if the current user is the owner and the real user is the createdByAgent. See an example at https://github.com/kgrgreer/foam3/blob/d5039484d241b9ee6643d432c9c1b39385e26b80/src/foam/nanos/ticket/Ticket.js#L351-L356.